std/algorithm/iteration.d

```diff
@@ -2948,6 +2955,15 @@ if (isInputRange!RoR && isInputRange!(ElementType!RoR))
     assert(!equal(js2, js));
 }
 
+// https://issues.dlang.org/show_bug.cgi?id=19213
+@system unittest
+{
+    auto results = [[1,2], [3,4]].map!(q => q.chunkBy!"a").joiner;
+    int i = 1;
+    foreach(ref e; results)
```

Comment:
Style target fails here due to missing whitespace after foreach. Why not use `equals`?
Also, we should probably test the bidirectional code too (use e.g. `retro`)

Reply:
The result is not a bidirectional range, so I can't use `retro`.